packages/drivers/local-driver/src/localDocumentStorageService.ts

```diff
@@ -99,26 +99,28 @@ export class LocalDocumentStorageService implements IDocumentStorageService {
 		const rawTree = await this.manager.getTree(versionId);
 		const snapshotTree = buildGitTreeHierarchy(rawTree, this.blobsShaCache, true);
 		const groupIds = new Set<string>(snapshotFetchOptions?.loadingGroupIds ?? []);
+		const attributesBlobId = snapshotTree.trees[".protocol"].blobs.attributes;
+		// Only populate contents for the blobs which are supposed to be returned.
+		const blobContents = new Map<string, ArrayBuffer>();
+		const attributesBlobData = await this.readBlob(attributesBlobId);
 		if (groupIds.has("") || groupIds.size === 0) {
 			// If the root is in the groupIds, we don't need to filter the tree.
-			// We can just strip the tree of all groupIds.
-			// If we want to include the root groupId,
-			await this.stripTreeOfMissingLoadingGroupIds(snapshotTree, groupIds);
+			// We can just strip the  of all groupIds as in collect the blobIds so that we can
+			// return blob contents only for those ids.
+			await this.stripTreeOfMissingLoadingGroupIds(snapshotTree, groupIds, blobContents);
 		} else {
 			const hasFoundTree = await this.filterTreeByLoadingGroupIds(
```

Comment:
I wonder if this algorithm is sufficient.

Reply:
Let's keep them separate for now. This one is already a little complicated, adding other functionality to it will make it a bit more complicated.